src/connectdlg.cpp

```diff
@@ -148,7 +153,7 @@ CConnectDlg::CConnectDlg ( CClientSettings* pNSetP, const bool bNewShowCompleteR
     if ( bShowCompleteRegList )
     {
         lvwServers->setSortingEnabled ( true );
-        lvwServers->sortItems ( 0, Qt::AscendingOrder );
+        lvwServers->sortItems ( LVC_NAME, Qt::AscendingOrder );
```

Comment:
The code you quoted is only used when the undocumented option `--showallservers` is given. In this case, no automatic ping sorting is done, and instead the display can be sorted by clicking on the column headers. The default sort is by name, set by the line above.

If `--showallservers` is not given, then column header sorting is not enabled, and the sorting by ping is done within `CConnectDlg::SetPingTimeAndNumClientsResult()` in these lines:

```
        // Update sorting. Note that the sorting must be the last action for the
        // current item since the topLevelItem(iIdx) is then no longer valid.
        // To avoid that the list is sorted shortly before a double click (which
        // could lead to connecting an incorrect server) the sorting is disabled
        // as long as the mouse is over the list (but it is not disabled for the
        // initial timer of about 2s, see TimerInitialSort) (#293).
        if ( bDoSorting && !bShowCompleteRegList &&
             ( TimerInitialSort.isActive() || !lvwServers->underMouse() ) ) // do not sort if "show all servers"
        {
            lvwServers->sortByColumn ( LVC_PING_MIN_HIDDEN, Qt::AscendingOrder );
        }
```
`LVC_PING_MIN_HIDDEN` is a hidden column used to store the minimum ping time received for each client since the connect dialog was opened.